src/classes/RD2_RecurringDonations_TEST.cls

```diff
@@ -162,6 +163,64 @@ private with sharing class RD2_RecurringDonations_TEST {
             'The triggerOld parameter should contain the data from the oldList parameter');
     }
 
+    /***
+    * @description Verifies schedule is created when recurring donation is inserted
+    */
+    @isTest
+    private static void shouldCreateNewRDSchedulesOnInsert() {
```

Comment:
This is works as an integration test, but I think you should add a new test class for `RD2_ScheduleService_TEST` that has unit tests methods to validate the actual functionality more granularly. You could change this method to only insert one RD and just validate that the trigger runs (i.e., the integration), but then have individual methods in the service test class that tests each individual type to validate that the schedule is created correctly.